src/pages/gen2/start/quickstart/vite-react-app/index.mdx

```diff
@@ -0,0 +1,270 @@
+import type { Schema } from "../../amplify/data/resource";
+
+export default function TodoList() {
+  // generate your data client using the Schema from your backend
+  const client = generateClient<Schema>();
+
+  const [todos, setTodos] = useState<Schema["Todo"][]>([]);
+
+  async function listTodos() {
+    // fetch all todos
+    const { data } = await client.models.Todo.list();
+    setTodos(data);
+  }
+
+  useEffect(() => {
+    listTodos();
+  }, []);
+
+  return (
+    <div>
+      <h1>Todos</h1>
+
+      <ul>
+        {todos.map((todo) => (
+          <li key={todo.id}>{todo.content}</li>
+        ))}
+      </ul>
+    </div>
+  );
+}
+```
+
+In `src/App.tsx`, import `components/TodoList` and render it:
+
+```ts title="src/App.tsx"
+import { withAuthenticator } from "@aws-amplify/ui-react";
+import "@aws-amplify/ui-react/styles.css";
+import TodoList from "./components/TodoList";
+
+function App() {
+  return (
+    <>
+      <h1>Hello, Amplify 👋</h1>
+      <TodoList />
+    </>
+  );
+}
+
+export default withAuthenticator(App);
+```
+
+Once you save the file and navigate back to `http://localhost:5173`, you should see a blank page for now because you have only an empty list of to-dos.
+
+### Create a new to-do item
+
+Let's update the return statement of the `components/TodoList` component to have a button for creating a new to-do list item, prompting the user to add the title and description. In a production app, you would want to create a full form, and run the create method on form submission.
+
+```tsx title="components/TodoList.tsx"
+// ...
+  return (
+    <div>
+      <h1>Todos</h1>
+      <button onClick={async () => {
+        // create a new Todo with the following attributes
+        const { errors, data: newTodo } = await client.models.Todo.create({
+          // prompt the user to enter the title
+          content: window.prompt("title"),
+          done: false,
+          priority: 'medium'
+        })
+        console.log(errors, newTodo);
+      }}>Create</button>
+
+      <ul>
+        {todos.map((todo) => (
+          <li key={todo.id}>{todo.content}</li>
+        ))}
+      </ul>
+    </div>
+  );
+}
+```
+
+Create a couple of to-dos, then refresh the page to see them. You can also subscribe to new to-dos in your `useEffect` to have them live reload on the page.
+
+```js
+useEffect(() => {
+  const sub = client.models.Todo.observeQuery().subscribe(({ items }) =>
+   setTodos([...items])
+  );
+
+  return () => sub.unsubscribe();
+}, []);
+```
+
+### Terminate dev server
+
+Go to `localhost` in the browser to make sure you can now log in and create and list to-dos. You can end your development session by shutting down the frontend dev server and cloud sandbox. The sandbox prompts you to delete your backend resources. While you can retain your backend, we recommend deleting all resources so you can start clean again next time.
+
+## Deploy and host a fullstack branch
```

Comment:
@chrisbonifacio  There is a fragment for this content - https://github.com/aws-amplify/docs/blob/main/src/fragments/gen2/quickstart/deploy-and-host.mdx

Reply:
It has some content specific to Next.js like the callout

The content I pulled from it might still need some other things verified and/or removed as I'm not sure if they're necessary for a Vite + React app such as the mention of the build image

Reply:
Yeah that's the problem with what we've tried to make into fragments.  It's not general enough to use everywhere and making more fragments is just more confusing.